### PR TITLE
Gce plume

### PR DESCRIPTION
**Pull request by pbx0**
This adds the all the basic commands necessary to create and destroy clusters on GCE. 

Most commands have extra optional parameters, but relying on default flag usage looks like
`plume upload`
`plume create-instances -image=<imagename> -n 3`
`plume list-instances`
`plume destroy-instances -prefix=<$USER>`
`plume list-images`

`create-instances` would in most cases specify a cloud-config filepath.

I prefixed the subcommand flag variables to avoid confusion and collision. Alternative approaches would be moving each subcommand into a separate package (but still having to register them in `main`) or reworking the way the CLI works.

gce.go will eventually be migrated into platform for use with kola as well.

### CODE REVIEW

File: cmd/plume/create.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	for _, vm := range vms {
+		fmt.Printf("To access %v use cmd:\n", vm.name)
+		fmt.Printf("ssh -o UserKnownHostsFile=/dev/null -o CheckHostIP=no -o StrictHostKeyChecking=no core@%v\n", vm.extIP)
+	}
+
+	return 0
+}
+
+// nextName returns the next available numbered name or the given base name.
+// snipped from: https://github.com/golang/build/blob/master/cmd/gomote/list.go
```
Comment (marineam) [inline cmd/plume/create.go]: As such perhaps fair to include the origional copyright in the header?

File: cmd/plume/gce.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+		if !strings.HasPrefix(image.Name, prefix) {
+			continue
+		}
+		images = append(images, image.Name)
+	}
+	return images, nil
+}
+
+// create gce VM and wait for creation to succeed. Some code snipped from:
+// https://github.com/golang/build/blob/master/buildlet/gce.go#L323
```
Comment (marineam) [inline cmd/plume/gce.go]: This is certainly big enough to deserve preserving the copyright notice.

**Comment by marineam** [comment]
Looks reasonable, my only concern is that there is enough copied code that the original copyright notice should be preserved, as required by the BSD license the code was released under.

**Comment by philips** [comment]
cc @kelseyhightower

File: cmd/plume/create.go (hunk 1 above)
Comment (pbx0) [inline cmd/plume/create.go]: So just add the Go author copyright up with ours? Should I still mention which functions have the code and a link to where it is from?

Comment (marineam) [inline cmd/plume/create.go]: Yes, add the same go author copyright to the top. Also keep the existing
comments/links. That is useful to know.

**Comment by pbx0** [comment]
Added the Go author copyright header and BSD license under the the file LICENSE-BSD next to our apache license file.

**Comment by marineam** [comment]
Relicensing BSD code under Apache-2 is perfectly fine so you just need to preserve the copyright line. Having both license headers in the file is just confusing.

**Comment by pbx0** [comment]
BSD license removed.

File: cmd/plume/gce.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
```
Comment (marineam) [inline cmd/plume/gce.go]: This is the bit I was referring to about having two license headers. Just need something like:

```
// Copyright 2015 CoreOS, Inc.
// Copyright 2015 The Go Authors.
//
// Licensed under the Apache License.....
```

Comment (pbx0) [inline cmd/plume/gce.go]: Makes more sense, fixed.

**Comment by marineam** [comment]
LGTM

**Pull request merged**